src/components/App/App.js

```diff
@@ -0,0 +1,153 @@
+import Header from '../Header/Header';
+import Home from '../Home/Home';
+import Character from '../Character/Character';
+import Main from '../Main/Main';
+import About from '../About/About';
+import Footer from '../Footer/Footer';
+import api from '../../utils/DAndDApi';
+import { useState, useEffect } from 'react';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import CardDetailsPopup from '../CardDetailsPopup/CardDetailsPopup';
+import Preloader from '../Preloader/Preloader';
+import './App.css';
+
+function App() {
+  const [cards, setCards] = useState([]);
+  const [classes, setClasses] = useState([]);
+  const [races, setRaces] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedCard, setSelectedCard] = useState(null)
```

Comment:
Caso queira iniciar com um objeto vazio ou nulo, é sempre melhor fazer:

- `useState({})`

Para declarar um objeto explicitamente vazio

Reply:
Opa Edu! Quando início este objeto vazio eu recebo um erro no CardDetailsPopup: can't convert undefined to object

![image](https://github.com/dlaranjeirasilva/myded_campaign_frontend/assets/44902922/cf496e1a-aee7-4873-8d4f-c7bab36bc08e)